From a4bdea04b6f7effbc6385635ec37321aea0f6d7c Mon Sep 17 00:00:00 2001
From: Fan Du <fan.du@intel.com>
Date: Thu, 23 Jul 2015 11:37:41 -0400
Subject: [PATCH] pci {guest kernel}: set pci net class bar to 3

For pci net class is defined as 0x020000
By optimaztion, guess_bar_count set net class bar count to 2.
This will cause subsquent bar resource missing after lkvm
commit [1] extend virtio net bar resouces with MMIO bar.

[1] a463650caad6 ("kvm tools: pci: add MMIO interface to virtio-pci devices")

Signed-off-by: Fan Du <fan.du@intel.com>
---
 drivers/pci/probe.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pci/probe.c b/drivers/pci/probe.c
index 73d1b9f..f3489a1 100644
--- a/drivers/pci/probe.c
+++ b/drivers/pci/probe.c
@@ -1253,7 +1253,7 @@ static int guess_bar_count(int class)
         if (class == 0x068000)
             return 0;
         if (class == 0x020000)
-            return 2;
+            return 3;
         if (class == 0x010000)
             return 2;
         if (class == 0x00ff00)
-- 
1.8.3.1

